internal/helper/helper.go

```diff
@@ -284,12 +277,47 @@ func SafeQuantitySubtraction(orig, subtract quantity.Size) quantity.Size {
 	return orig - subtract
 }
 
+// CreateTarArchive places all of the files from a source directory into a tar.
+// Currently supported are uncompressed tar archives and the following
+// compression types: zip, gzip, xz bzip2, zstd
+func CreateTarArchive(src, dest, compression string, verbose, debug bool) error {
+	tarCommand := *exec.Command("tar",
+		"--directory",
+		src,
+		"--xattrs",
+		"--create",
+		"--file",
+		dest,
+		".",
+	)
+	fmt.Println(tarCommand.String())
+	if debug {
+		tarCommand.Args = append(tarCommand.Args, "--verbose")
+	}
+	tarOutput := SetCommandOutput(&tarCommand, debug)
+	if err := tarCommand.Run(); err != nil {
+		return fmt.Errorf("Error running \"tar\" command \"%s\". "+
+			"Error is \"%s\". Full output below:\n%s",
+			tarCommand.String(), err.Error(), tarOutput.String())
+	}
+	return nil
+}
+
 // ExtractTarArchive extracts all the files from a tar. Currently supported are
 // uncompressed tar archives and the following compression types: zip, gzip, xz
 // bzip2, zstd
 func ExtractTarArchive(src, dest string, verbose, debug bool) error {
-	// magic numbers are used to determine the compression type of
-	// the tar archive
+	tarCommand := *exec.Command("tar",
+		"--extract",
+		"--file",
+		src,
+		"--directory",
+		dest,
+	)
+	if debug {
+		tarCommand.Args = append(tarCommand.Args, "--verbose")
+	}
+	// determine if there is compression on the tar archive, and what kind
```

Comment:
I thought tar did this itself these days? Does it miss one of the format we want to support?

Reply:
It does miss regular .zip (not gzip). I've just asked the CPC team if they'd be okay with me dropping support for zip so we can get rid of this whole "magic number" thing.

On this topic, should we add in support for creating tarballs with `--lzip`, `--lzma`, `--lzop`, or `--compress`?

Reply:
> On this topic, should we add in support for creating tarballs with --lzip, --lzma, --lzop, or --compress?

Don't listen to me too much on this, but I'm going to go with "no". gzip, zstd and xz should be enough for everyone?